docs/state-diff/README.md

```diff
@@ -0,0 +1,185 @@
+
+On top of that, the `save_address` function allows to save the name of the deployed contract and their addresses into a separate JSON file. This will be crucial to easily write the symbolic property tests later on.
+
+### 🏷️ Name the files 🏷️
+
+There are two different JSON files that can be created, one containing the recorded state updates and another one with the saved names of important addresses. On top of that, we have to define in which directory do these files live. For simplicity, we've made it so that these parameters are set with environment variables:
+- `STATE_DIFF_NAME`: Name of the JSON containing the state updates. Example: `StateDiff.json`
+- `ADDR_NAMES`: Name of the JSON containing the saved names of relevant addresses. Example: `AddressNames.json`
+- `STATE_DIFF_DIR`: Path relative to the foundry root dir where the files will be stored. Example: `state-diff`
+
+### ⭕ Additional permissions ⭕
+
+Before executing the state-recording function you'll need to give Foundry permissions to write the JSON files in the specified directory. To give Foundry write permissions for these files you can add the following to the Foundry profile. Note that the `path` assignment has to be the same as the value set for the `STATE_DIFF_NAME` variable:
+```toml
+fs_permissions = [
+  { access='read-write, path='state-diff' }
+]
+```
+Not adding this would result in a `the path state-diff/StateDiff.json is not allowed to be accessed for write operations` error.
+
+### 🏃 Run the recording 🏃
+
+Run your function containing the initial set up of your proofs (`counterBed` or `counterBedNamed` in our [example](./test/kontrol/state-diff/proof-initialization.sol)) with `forge script state-diff/proof-initialization.sol:CounterBed --sig counterBed --ffi`. Running it with `forge test` will also work, but only if its name starts with `test`. Notice the `--ffi` flag: we use `mkdir` and `touch` to handle the cases where the state diff files don't yet exist.
+
+### 🧼 Clean the State Diff file 🧼
+
+Currently, the produced state diff JSON is escaped when generated. Run the [`clean_json.py`](test/kontrol/scripts/json/clean_json.py) script on the generated state diff file to unescape it. Example: `python3 test/kontrol/scripts/json/clean_json.py state-diff/StateDiff.json`
+
+### 📜 Scripting all of this 📜
+
+Phew! This is nothing short of a mouthful of steps! In case you don't want to generate the necessary scripts by manally repeating all that we have gone through, we provide the [`record-state-diff.sh`](test/kontrol/scripts/record-state-diff.sh) script automatically generate the JSON files.
```

Comment:
I'd just say
```suggestion
Phew! This is nothing short of a mouthful of steps! In case you don't want to manually repeat all that we have gone through, we provide the [`record-state-diff.sh`](test/kontrol/scripts/record-state-diff.sh) script automatically generate the JSON files.
```
Please feel free to ignore the suggestion! But please note the typo in "manually" :)

Reply:
Oh my!  It will not come the day I don't do typos :)